src/provider/model/Model.ts

```diff
@@ -351,7 +351,7 @@ export class Model {
         if (entityType === EntityType.EXTERNAL_CONNECTION) {
             // Any connections to the service from adapters should be immediately registered
             const appInfo = await this._environment.inferApplication(identity);
-            const liveApp = new LiveApp(undefined);
+            const liveApp = this._liveAppsByUuid[identity.uuid] || new LiveApp(undefined);
 
             liveApp.setAppInfo(appInfo);
             this._liveAppsByUuid[identity.uuid] = liveApp;
```

Comment:
Suggest check for existing live app first, and only if there isn't one, do `inferApplication`/`setAppInfo`